src/psyclone/tests/psyir/frontend/fparser2_common_block_test.py

```diff
@@ -0,0 +1,166 @@
+    UnknownFortranType
+
+
+@pytest.mark.usefixtures("f2008_parser")
+def test_named_common_block():
+    ''' Test that named common blocks are correctly captured and the symbols
+    they reference have a CommonBlockInterface. '''
+
+    # Create a dummy test routine
+    routine = Routine("test_routine")
+    symtab = routine.symbol_table
+    processor = Fparser2Reader()
+
+    # Test with a single parameter
+    reader = FortranStringReader('''
+        integer :: a, b, c
+        common /name1/ a, b, c''')
+    fparser2spec = Specification_Part(reader)
+    processor.process_declarations(routine, fparser2spec.content, [])
+
+    # There is a name1 commonblock symbol
+    commonblock = symtab.lookup("_PSYCLONE_COMMONBLOCK_name1")
+    assert isinstance(commonblock.datatype, UnknownFortranType)
+    assert commonblock.datatype.declaration == "COMMON /name1/ a, b, c"
+
+    # The variables have been updated to a common block interface
+    assert isinstance(symtab.lookup("a").interface, CommonBlockInterface)
+    assert isinstance(symtab.lookup("b").interface, CommonBlockInterface)
+    assert isinstance(symtab.lookup("c").interface, CommonBlockInterface)
+
+    # The same common block can also bring other variables in a separate
+    # statement
+    reader = FortranStringReader('''
+        real :: d, e
+        common /name1/ d, e''')
+    fparser2spec = Specification_Part(reader)
+    processor.process_declarations(routine, fparser2spec.content, [])
+
+    # This is stored in a separate symbol, but the declaration has the right
+    # text
+    commonblock_2 = symtab.lookup("_PSYCLONE_COMMONBLOCK_name1_1")
+    assert commonblock_2.datatype.declaration == "COMMON /name1/ d, e"
+    assert isinstance(symtab.lookup("d").interface, CommonBlockInterface)
+    assert isinstance(symtab.lookup("e").interface, CommonBlockInterface)
+
+
+@pytest.mark.usefixtures("f2008_parser")
+def test_unnamed_commonblock():
+    ''' Test that unnamed common blocks produce NotImplementedError '''
+
+    # Create a dummy test routine
+    routine = Routine("test_routine")
+    symtab = routine.symbol_table
+    processor = Fparser2Reader()
+
+    # And provide an unnamed common block
+    reader = FortranStringReader('''
+        integer :: a, b, c
+        common a, b, c''')
+    fparser2spec = Specification_Part(reader)
+    processor.process_declarations(routine, fparser2spec.content, [])
+
+    # There is an unamed commonblock symbol
+    commonblock = symtab.lookup("_PSYCLONE_COMMONBLOCK_unnamed")
+    assert isinstance(commonblock.datatype, UnknownFortranType)
+    assert commonblock.datatype.declaration == "COMMON // a, b, c"
+
+    # The variables have been updated to a common block interface
+    assert isinstance(symtab.lookup("a").interface, CommonBlockInterface)
+    assert isinstance(symtab.lookup("b").interface, CommonBlockInterface)
+    assert isinstance(symtab.lookup("c").interface, CommonBlockInterface)
```

Comment:
Please could you add a test for the case where we have a COMMON block named "unnamed".

Reply:
I didn't because I don't use the unnamed symbol identifier anymore.